pkg/crc/image/image.go

```diff
@@ -0,0 +1,112 @@
+package image
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"runtime"
+	"strings"
+
+	crcpreset "github.com/code-ready/crc/pkg/crc/preset"
+	"github.com/containers/image/v5/copy"
+	"github.com/containers/image/v5/signature"
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/containers/image/v5/types"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+var (
+	registryURI = "quay.io"
+	orgName     = "crcont"
+)
+
+type OCIImageHandler struct {
+	ImageName string
+	ImageTag  string
+	Progress  io.Writer
+}
+
+func (img *OCIImageHandler) uri() string {
+	return strings.Join([]string{registryURI, orgName, img.ImageName}, "/")
+}
+
+func (img *OCIImageHandler) parse(image string) (types.ImageReference, error) {
+	return alltransports.ParseImageName(image)
+}
+
+func (img *OCIImageHandler) policyContext() (*signature.PolicyContext, error) {
+	policy := &signature.Policy{Default: []signature.PolicyRequirement{signature.NewPRInsecureAcceptAnything()}}
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return nil, fmt.Errorf("error creating security context: %s", err.Error())
+	}
+
+	return policyContext, nil
+}
+
+func (img *OCIImageHandler) getSystemContext() *types.SystemContext {
+	return &types.SystemContext{
+		OSChoice: "linux",
+	}
+}
+
+// CopyImage pull the image from the registry and put it to local storage
```

Comment:
`local storage` means `destPath`? It could be misread as `podman local image s torage`, so I'd mention destpath in the doc.

Reply:
@cfergeau you mean `podman load <image.tar>` ?

Reply:
Also I will put destpath.